pkg/controller/controller.go

```diff
@@ -726,7 +727,8 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 	pvName := req.Name
 	provisionerCredentials := req.Secrets
 
-	createCtx, cancel := context.WithTimeout(ctx, p.timeout)
+	createCtx := markAsMigrated(ctx, result.migratedVolume)
```

Comment:
is it possible to add/extend a unit test to check that we set the ctx correctly?

Reply:
I tried to capture the Context argument that is being passed to CreateVolume by using this logic:

controllerServerMock.EXPECT().CreateVolume(/*arguments*/).Do(func(ctx context.Context, req *csi.CreateVolumeRequest) {
						capturedContext = ctx
				})

and then validating the captured context using below logic:

			if (capturedContext == nil || capturedContext.Value(connection.AdditionalInfoKey) == nil) {
				t.Errorf("Got no migrated label in context, expected migrated label")
			}

But the context that is being captured by 'Do' doesn't contain the values for the AdditionalInfoKey or any other key that I tried to set and validate.

Please let me know if you can know of any other way to do it.

Reply:
Can we add a unit test for the markAsMigrated function?

Reply:
Done